src/github/SyftGithubAction.ts

```diff
@@ -0,0 +1,398 @@
+  const { repo } = github.context;
+  const client = getClient(repo, core.getInput("github_token"));
+
+  const fileName = getArtifactName();
+
+  const tempPath = fs.mkdtempSync(path.join(os.tmpdir(), "sbom-action-"));
+  const filePath = `${tempPath}/${fileName}`;
+  fs.writeFileSync(filePath, contents);
+
+  const outputFile = core.getInput("output_file");
+  if (outputFile) {
+    fs.copyFileSync(filePath, outputFile);
+  }
+
+  core.info(dashWrap(`Uploading workflow artifacts`));
+  core.info(`Artifact: ${filePath}`);
+
+  await client.uploadWorkflowArtifact({
+    file: filePath,
+    name: fileName,
+  });
+}
+
+/**
+ * Gets a boolean input value if supplied, otherwise returns the default
+ * @param name name of the input
+ * @param defaultValue default value to return if not set
+ */
+function getBooleanInput(name: string, defaultValue: boolean): boolean {
+  const val = core.getInput(name);
+  if (val === "") {
+    return defaultValue;
+  }
+  return Boolean(val);
+}
+
+export async function runSyftAction(): Promise<void> {
+  try {
+    core.info(dashWrap(`Running SBOM Action`));
+
+    core.debug(`Got github context:`);
+    core.debug(JSON.stringify(github.context));
+
+    const start = Date.now();
+
+    const doUpload = getBooleanInput("upload_artifact", true);
+    const comparePulls = getBooleanInput("compare_pulls", false);
+    const outputVariable = core.getInput("output_var");
+
+    const output = await executeSyft({
+      input: {
+        path: core.getInput("path"),
+        image: core.getInput("image"),
+      },
+      format: getSbomFormat(),
+    });
+
+    core.info(`SBOM scan completed in: ${(Date.now() - start) / 1000}s`);
+
+    if (output) {
+      const { eventName, payload, repo } = github.context;
+      if (comparePulls && eventName === "pull_request") {
+        const client = getClient(repo, core.getInput("github_token"));
+
+        const pr = (payload as PullRequestEvent).pull_request;
+        const branchWorkflow = await client.findLatestWorkflowRunForBranch({
+          branch: pr.base.ref,
+        });
+
+        core.debug("Got branchWorkflow");
+        core.debug(JSON.stringify(branchWorkflow));
+
+        if (branchWorkflow) {
+          const baseBranchArtifacts = await client.listWorkflowRunArtifacts({
+            runId: branchWorkflow.id,
+          });
+
+          core.debug("Got baseBranchArtifacts");
+          core.debug(JSON.stringify(baseBranchArtifacts));
+
+          for (const artifact of baseBranchArtifacts) {
+            if (artifact.name === getArtifactName()) {
+              const baseArtifact = await client.downloadWorkflowRunArtifact({
+                artifactId: artifact.id,
+              });
+
+              core.info(
+                `Downloaded SBOM from ${pr.base.ref} to ${baseArtifact}`
+              );
+            }
+          }
+        }
+      }
+
+      const priorArtifact = process.env[PRIOR_ARTIFACT_ENV_VAR];
```

Comment:
Is this just so the logs can show the user what the PRIOR_ARTIFACT_ENV_VAR was before going with an upload? I don't see it used anywhere else.

Reply:
Yes. It could be deleted, but I thought it might end up being useful if we wanted to chain the entire pipeline together at some point. I was mainly trying to see if it was possible to do something like that.

Reply:
... i've made this just print in debug mode for now